File: lecture_6/assignements/sentimental-cash/cash.py
from cs50 import get_float


COINS = [25, 10, 5, 1]


def main():
    change = get_valid_change()
    print(coin_counter(change))


def get_valid_change():
    while True:
        change = get_float("Change: ")

        if change > 0:
            return (change * 100)


def coin_counter(n):
    coins = 0

    for m in COINS:
        while n > (m - 1):
            n -= m
            coins += 1

    return coins


main()

File: lecture_0-2/assignements/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int count_coins(int change, int coin_value);

// Define constant values of coins
const int quarter = 25;
const int dime = 10;
const int nickel = 5;
const int penny = 1;

int main(void)
{

    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int quarters = count_coins(cents, quarter);
    cents = cents - (quarter * quarters);

    int dimes = count_coins(cents, dime);
    cents -= (dime * dimes);

    int nickels = count_coins(cents, nickel);
    cents -= (nickel * nickels);

    int pennies = count_coins(cents, penny);

    int sum = quarters + dimes + nickels + pennies;

    printf("%i\n", sum);

    return 0;
}

int count_coins(int change, int coin_value)
{
    int coin = 0;
    while (change >= coin_value)
    {
        coin++;
        change -= coin_value;
    }
    return coin;
}
